src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9805,12 +9805,70 @@ private ArgumentSyntax ParseArgumentExpression(bool isIndexer)
                 else
                 {
                     expression = this.ParseSubExpression(Precedence.Expression);
+
+                    if (refOrOutKeyword?.Kind == SyntaxKind.OutKeyword &&
+                        expression.Kind == SyntaxKind.InvocationExpression)
+                    {
+                        var invocation = (InvocationExpressionSyntax)expression;
+                        ExpressionSyntax invocationTarget = invocation.Expression;
+
+                        // See if the expression is an invocation that could also be successfully parsed and interpreted
+                        // as a deconstruction target. I.e. something like "var (x, y)" or "var (x, (y, z))".
+                        // We should report an error in this case because we plan to support deconstruction for out arguments
+                        // in the future. We need to ensure that we do not successfully interpret this as an invocation of a
+                        // ref-returning method named var with two, or more arguments. 
+
+                        if (invocationTarget.Kind == SyntaxKind.IdentifierName &&
+                            ((IdentifierNameSyntax)invocationTarget).Identifier.IsVar() &&
```

Comment:
I would be tempted to stop there. We just don't allow "var", regardless of how many arguments or if the argument list could look like a deconstruction.